src/main/java/bio/terra/janitor/service/janitor/JanitorService.java

```diff
@@ -126,11 +127,72 @@ public TrackedResourceInfoList getResources(
         janitorDao.retrieveResourcesWith(cloudResourceUid);
     TrackedResourceInfoList resourceList = new TrackedResourceInfoList();
     resourcesWithLabels.stream()
-        .map(resourceWithLabels -> createInfo(resourceWithLabels))
+        .map(JanitorService::createInfo)
         .forEach(resourceList::addResourcesItem);
     return resourceList;
   }
 
+  /**
+   * Updates the resource state.
+   *
+   * <p>Currently it supports:
+   *
+   * <ul>
+   *   <li>Abandon resource: Update resource from READY or CLEANING to ABANDONED
+   *   <li>Bump resource: Update resource from ABANDONED to READY
+   * </ul>
+   */
+  public void updateResource(
+      CloudResourceUid cloudResourceUid, ResourceState state, AuthenticatedUserRequest userReq) {
+    iamService.requireAdminUser(userReq);
+    if (state == ResourceState.ABANDONED) {
+      // Update state to ABANDONED for READY or CLEANING state resources.
+      List<TrackedResource> resources =
+          getResourceWithState(
+              cloudResourceUid, TrackedResourceState.READY, TrackedResourceState.CLEANING);
+
+      if (resources.size() > 1) {
+        throw new InternalServerErrorException(
+            "More than one READY or CLEANING state resources are found.");
+      }
+      janitorDao.updateResourceState(
+          resources.get(0).trackedResourceId(), TrackedResourceState.ABANDONED);
+    } else if (state == ResourceState.READY) {
+      // Bump ABANDONED state resources for cleaning.
+      List<TrackedResource> resources =
+          getResourceWithState(cloudResourceUid, TrackedResourceState.ABANDONED);
+
+      // It is possible that there might be multiple ABANDON resources, and we only need to update
+      // the one with last
+      // expiration time.
+      TrackedResource latestResource =
+          resources.stream().max(Comparator.comparing(TrackedResource::expiration)).get();
+      janitorDao.updateResourceState(
+          latestResource.trackedResourceId(), TrackedResourceState.READY);
+    }
```

Comment:
If we add additional ResourceState enum values in the future, it would be easy to forget to add another conditional block here. Maybe add a catch-all else statement which returns an internal server error?

Reply:
Done!